server/sonar-server/src/main/java/org/sonar/server/issue/notification/NewIssuesNotification.java

```diff
@@ -39,39 +42,42 @@ public NewIssuesNotification() {
   }
 
   public NewIssuesNotification setAnalysisDate(Date d) {
-    setFieldValue("projectDate", DateUtils.formatDateTime(d));
+    setFieldValue(FIELD_PROJECT_DATE, DateUtils.formatDateTime(d));
     return this;
   }
 
   public NewIssuesNotification setProject(ComponentDto project) {
-    setFieldValue("projectName", project.longName());
-    setFieldValue("projectKey", project.key());
-    setFieldValue("projectUuid", project.uuid());
+    setFieldValue(FIELD_PROJECT_NAME, project.longName());
+    setFieldValue(FIELD_PROJECT_KEY, project.key());
+    setFieldValue(FIELD_PROJECT_UUID, project.uuid());
     return this;
   }
 
-  public NewIssuesNotification setStatistics(Component project, Stats stats) {
-    setDefaultMessage(stats.size() + " new issues on " + project.longName() + ".\n");
-    setFieldValue("count", String.valueOf(stats.size()));
-    for (String severity : Severity.ALL) {
-      setFieldValue("count-" + severity, String.valueOf(stats.countIssuesWithSeverity(severity)));
-    }
-    return this;
-  }
+  public NewIssuesNotification setStatistics(Component project, NewIssuesStatistics stats) {
+    setDefaultMessage(stats.countForMetric(SEVERITY) + " new issues on " + project.longName() + ".\n");
 
-  public static class Stats {
-    private final Multiset<String> set = HashMultiset.create();
+    setSeverityStatistics(stats);
+    setTop5CountsForMetric(stats, METRIC.LOGIN);
+    setTop5CountsForMetric(stats, METRIC.TAGS);
+    setTop5CountsForMetric(stats, METRIC.COMPONENT);
+    int hoursInDay = 24;
```

Comment:
Default value is 8 and it can be overridden in general settings. @julienlancelot do you confirm ? What's the best approach ?